src/main/java/hudson/plugins/plot/PlotBuilder.java

```diff
@@ -34,133 +37,188 @@
  */
 public class PlotBuilder extends Builder implements SimpleBuildStep {
 
+    // Required fields
     private final String group;
-    private final String title;
-    private final String numBuilds;
-    private final String yaxis;
     private final String style;
-    private final boolean useDescr;
-    private final boolean exclZero;
-    private final boolean logarithmic;
-    private final boolean keepRecords;
-    private final String yaxisMinimum;
-    private final String yaxisMaximum;
+
+    // Optional fields
+    @CheckForNull
+    private String title;
+    @CheckForNull
+    private String numBuilds;
+    @CheckForNull
+    private String yaxis;
+    @CheckForNull
+    private String yaxisMinimum;
+    @CheckForNull
+    private String yaxisMaximum;
+    private boolean useDescr;
+    private boolean exclZero;
+    private boolean logarithmic;
+    private boolean keepRecords;
+
+    // Generated?
     @SuppressWarnings("visibilitymodifier")
     public String csvFileName;
```

Comment:
This one is a little odd, since it's not directly specified by the user with the snippet generator. I'm assuming it's generated [here](https://github.com/rmstyrczula/plot-plugin/blob/1a7a8de08b54f03a10c961bd0b2fc8f952a17acc/src/main/java/hudson/plugins/plot/PlotBuilder.java#L264) through reflection when the step is created.

Reply:
No need to move, your assumption is correct. Maybe, in this case, explanation comment from where it comes will be better than `Generate?`.